.travis.yml

```diff
@@ -13,22 +13,49 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# Travis CI build file for Swift gRPC.
+# Travis CI build file for gRPC Swift.
 
-os:
-  - linux
-  - osx
+jobs:
+  include:
+    # Tests for each PR.
+    - &tests
+      stage: "Test"
+      name: "Ubuntu 18.04"
+      os: linux
+      dist: bionic
+      script: ./.travis-script.sh
+      env: RUN_INTEROP_TESTS=false
+    - <<: *tests
+      name: "macOS (Xcode 10.3)"
+      os: osx
+      osx_image: xcode10.3
+    # Interop Tests.
+    - &interop_tests
+      stage: "Interoperability Tests"
+      # Linux
+      name: "Ubuntu 18.04"
+      os: linux
+      dist: bionic
+      script: ./.travis-script.sh
+      env: RUN_INTEROP_TESTS=true
+    - <<: *interop_tests
+      name: "macOS (Xcode 10.3)"
+      os: osx
+      osx_image: xcode10.3
+
+stages:
+  # Only run stage for PRs
```

Comment:
What other types are there? Should we maybe _always_ run tests?

Reply:
It's only: `push`, `pull_request`, `api`, and `cron` (https://docs.travis-ci.com/user/conditions-v1#integration). 

Actually running them when we merge is a good idea: a clean merge doesn't necessarily mean the outcome was correct.

Reply:
Should we run this for both pushes and PRs then?